sdk/cosmos/azure-cosmos-encryption/src/main/java/com/azure/cosmos/CosmosEncryptionAsyncClient.java

```diff
@@ -0,0 +1,147 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.cosmos;
+
+import com.azure.cosmos.implementation.caches.AsyncCache;
+import com.azure.cosmos.models.ClientEncryptionPolicy;
+import com.azure.cosmos.models.CosmosClientEncryptionKeyProperties;
+import com.microsoft.data.encryption.cryptography.EncryptionKeyStoreProvider;
+import reactor.core.publisher.Mono;
+
+/**
+ * CosmosClient with Encryption support.
+ */
+public class CosmosEncryptionAsyncClient {
+    private final CosmosAsyncClient cosmosAsyncClient;
+    private final AsyncCache<String, ClientEncryptionPolicy> clientEncryptionPolicyCacheByContainerId;
+    private final AsyncCache<String, CosmosClientEncryptionKeyProperties> clientEncryptionKeyPropertiesCacheByKeyId;
+    private EncryptionKeyStoreProvider encryptionKeyStoreProvider;
+
+    CosmosEncryptionAsyncClient(CosmosAsyncClient cosmosAsyncClient,
+                                EncryptionKeyStoreProvider encryptionKeyStoreProvider) {
+        if (cosmosAsyncClient == null) {
+            throw new IllegalArgumentException("cosmosClient is null");
+        }
+        if (encryptionKeyStoreProvider == null) {
+            throw new IllegalArgumentException("encryptionKeyStoreProvider is null");
+        }
+        this.cosmosAsyncClient = cosmosAsyncClient;
+        this.encryptionKeyStoreProvider = encryptionKeyStoreProvider;
+        this.clientEncryptionKeyPropertiesCacheByKeyId = new AsyncCache<>();
+        this.clientEncryptionPolicyCacheByContainerId = new AsyncCache<>();
+    }
+
+    public EncryptionKeyStoreProvider getEncryptionKeyStoreProvider() {
+        return encryptionKeyStoreProvider;
+    }
+
+    Mono<ClientEncryptionPolicy> getClientEncryptionPolicyAsync(
+        CosmosAsyncContainer container,
+        boolean shouldForceRefresh) {
+        // container Id is unique within a Database.
+        String cacheKey =
+            container.getDatabase().getId() + "/" + container.getId();
+
+        // cache it against Database and Container ID key.
+        if (!shouldForceRefresh) {
+            return this.clientEncryptionPolicyCacheByContainerId.getAsync(
+                cacheKey,
+                null,
+                () -> container.read().map(cosmosContainerResponse -> cosmosContainerResponse.getProperties().getClientEncryptionPolicy()));
```

Comment:
could you please make one method call per line here and in other methods in this class?

it is hard to read